### PR TITLE
Window function frame clause & refactor

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

File: common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala [hunk 1]
```diff
@@ -159,6 +160,7 @@ object SqlFunctions extends SqlFunctionsLocation with SqlFunctionsGeometry with
     Rank -> nary("rank") _,
     DenseRank -> nary("dense_rank") _,
     FirstValue -> nary("first_value") _,
+    LastValue -> nary("last_value") _,
     // LastValue -> nary("last_value") _, TODO: consider adding frame clause support before supporting this
```
Comment (catstavi) [inline common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala:164]: maybe this commented out version of the line above should be removed now?

File: common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala [hunk 2]
```diff
@@ -360,6 +362,44 @@ object SqlFunctions extends SqlFunctionsLocation with SqlFunctionsGeometry with
...
+          (acc._1 :+ sql, newSetParams)
+        }
+
+        val sqlOrderings = windowFunctionInfo.orderings.foldLeft((Seq.empty[String], sqlPartitions._2)) { (acc, ob) =>
+          val ParametricSql(Seq(sql), newSetParams) = Sqlizer.sql(ob)(rep, typeRep, acc._2, ctx, escape)
+          (acc._1 :+ sql, newSetParams)
+        }
+
+        val sqlFrames = windowFunctionInfo.frames.foldLeft((Seq.empty[String], sqlOrderings._2)) { (acc, ob) =>
+          Sqlizer.sql(ob)(rep, typeRep, acc._2, ctx, escape)
```
Comment (catstavi) [inline common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala:386]: what's going on here?

Comment (chitang) [reply common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala:386]: i must be missing something, looking

Comment (chitang) [reply common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala]: yep, i forgot that i had not finished while i was waiting for soql-reference.  pushed a commit to fix this.

**Pull request merged**